### ISSUE TITLE
Setup role based authentication and authorization for Airflow UI with Cognito

### ISSUE THREAD

**Issue by ramesh-maddegoda** [NASA-PDS/nucleus#102]
### 💡 Description

At the moment, those who have access to the MCP AWS account can use many functionalities of MWAA. It is required restrict the features based on roles of users.

**Comment by jordanpadams** [comment]
Status: Tested with 2 different roles but issues with MCP roles. To follow with MCP regarding adding/remove Admin. Task to document how to set this up.

**Comment by nutjob4life** [comment]
Had a chat with Gabe from MCP, considering temporary users to test with various roles (because we always have the admin role). Need the extra roles to test each situation.

**Comment by tloubrieu-jpl** [comment]
To be discussed with Gabe when Ramesh is back.

**Comment by tloubrieu-jpl** [comment]
Ramesh looks at Cognito authentication as an alternative to solve the issue. That allows to attach specific IAM policies to users which match with their actual authorization.

**Comment by ramesh-maddegoda** [comment]
Tested COgnito integration with the approach proposed with the following Amazon articles.

Accessing a private Amazon MWAA environment using federated identities
https://d1.awsstatic.com/whitepapers/accessing-a-private-amazon-mwaa-environment-using-federated-identities.pdf

Application load balancer single-sign-on for Amazon MWAA
https://github.com/aws-samples/alb-sso-mwaa

There was a problem related with validating the digital signature of the access token. The documentation says, that issue will not be there, when the lambda zip file is built on a linux system with required cryptography libraries. 

Regardless of that, it was possible to test the MWAA Role-based access with Cognito users.

**Comment by ramesh-maddegoda** [comment]
If we do not want to use Cognito for Nucleus users, another option available creating AWS accounts for user, create MWAA IAM for those users such as View, Ops, Admin etc. This option is much easier to implement technically, compared to the Cognito option.

**Comment by tloubrieu-jpl** [comment]
@ramesh-maddegoda need to present the architecture to the JPL SA's to validate it for production.

**Comment by tloubrieu-jpl** [comment]
@ramesh-maddegoda provided the architecture proposed to the SAs.

**Comment by tloubrieu-jpl** [comment]
We decided to move forward with the Cognito based authentication.

**Comment by tloubrieu-jpl** [comment]
The SSO works but the redirection between nucleus and cognito.

**Comment by tloubrieu-jpl** [comment]
@ramesh-maddegoda is still investigating the redirection issue.

**Comment by jordanpadams** [comment]
Standup status: blocked by intermittent issues with Cognito / Lambda solution for authentication.

**Comment by ramesh-maddegoda** [comment]
Created pull request https://github.com/NASA-PDS/nucleus/pull/122

**Comment by ramesh-maddegoda** [comment]
The pull request https://github.com/NASA-PDS/nucleus/pull/122 was merged.

As a part of this ticket, an ALB based approach to enable Cognito authentication for Nucleus Airflow UI was implemented. However, the ALB based approach only worked sometimes and currently there is an Amazon support ticket Case 172781777100323) to troubleshoot it. At the moment, [a python script i](https://github.com/NASA-PDS/nucleus/blob/main/utils/get-airflow-ui-webtoken.py)s used to get a web token URL to access Airflow UI.

A new ticket [Web based Cognito authentication for Nucleus Airflow UI](https://github.com/NASA-PDS/nucleus/issues/123) was created to focus on resolving this ALB related problem and eventually implement web based Cognito authentication for Nucleus Airflow UI.
